agent/connect_ca_endpoint_test.go

```diff
@@ -24,16 +24,16 @@ func TestConnectCARoots_empty(t *testing.T) {
 
 	t.Parallel()
 
-	require := require.New(t)
+	r := require.New(t)
```

Comment:
sadly I think these changes to require/assert may prevent `lint-consul-retry` from catching the problem it is supposed to catch.

Looking at [this line](https://github.com/hashicorp/lint-consul-retry/blob/master/lint-consul-retry.go#L189-L191) it seems to expect the name `require` or `assert` (the var name `pkg` is a bit misleading, it's just the identifier.

The way I generally fix the "shadowing" warning (which is I assume the motivation for these changes) is to remove the `require.New` or `assert.New`. IMO, storing `t` in the struct ends up leading to subtests using the wrong `testing.T` and doesn't really save much typing, or make anything significantly more readable. The consistency of always having the `t` there is probably better.

Not a blocker, but probably not a pattern we should continue to use.

Longer term I'd like to fix the `retry.Run` interface to be more like https://pkg.go.dev/gotest.tools/v3/poll, which removes the need for a linter, because accidentally using the wrong `testing.T` is no longer possible.

Reply:
I agree I will revert the fix for the time being